File: Goodstein/src/com/priorsoftware/goodstein/TestSequence.java
package com.priorsoftware.goodstein;

public class TestSequence {

	public static void main(final String[] args) {
		final int arg = Integer.valueOf(args[0]);
		final OrderedSum seq = OrderedSum.getBinaryExpansion(arg);
		System.out.println(seq);
		final double orig = seq.valueAt(2);
		System.out.println(orig);
	}
}

File: GoodsteinScala/src/com/priorsoftware/goodstein/scala/TestSequence.scala
package com.priorsoftware.goodstein.scala

object TestSequence {

  def main(args: Array[String]): Unit = {
    val arg = args(0).toInt
    val seq = OrderedSum.getBinaryExpansion(arg.intValue);
    println(seq);
    val orig = seq.valueAt(2);
    println(orig);
  }
}